### ISSUE TITLE
Windows Build

### ISSUE THREAD

**Issue by spitfiredd** [wamuir/fpds-conversion-utility#1]
Is it possible to build this on Windows 10?

**Comment by wamuir** [comment]
Yes, there should be a few ways to accomplish that.   I would need to look into this, as I am not a Windows user.

I could add a docker script to get things going for you in the short term, assuming you have access to docker.  Would that work for you?

**Comment by spitfiredd** [comment]
A docker script would work for me, I was trying to convert this to python but I am running into issues when I normalized the XML.

For example:

```
import lxml.etree as et

norm_rec_file = "normalize-record.xsl"
xml_file = "testIDV.xml"

raw_xml = et.parse(xml_file)
transform = et.XSLT(et.parse(norm_rec_file))
normed_xml = transform(raw_xml)
```

This just returns an empty string...

**Comment by spitfiredd** [comment]
I also tried to create a docker script but I am not really good at building code and this failed to compile.

```
FROM ubuntu:16.04

RUN apt-get update \
    && apt-get install -yq --no-install-recommends \
        build-essentials \
        make \
        cmake \ 
        libsqlite3-dev \
        libxml2 \
        libxslt1-dev \
        uuid-dev \

WORKDIR /usr/local/bin

COPY . ./
RUN mkdir build && cmake -S fpds-conversion-utility -B build && cmake --build build
```

**Comment by wamuir** [comment]
> A docker script would work for me, I was trying to convert this to python but I am running into issues when I normalized the XML.
> 
> For example:
> 
> ```
> import lxml.etree as et
> 
> norm_rec_file = "normalize-record.xsl"
> xml_file = "testIDV.xml"
> 
> raw_xml = et.parse(xml_file)
> transform = et.XSLT(et.parse(norm_rec_file))
> normed_xml = transform(raw_xml)
> ```
> 
> This just returns an empty string...


You'll need to iterate:

```python
import lxml.etree as et

norm_rec_file = "normalize-record.xsl"
xml_file = "testIDV.xml"


# raw_xml = et.parse(xml_file)
transform = et.XSLT(et.parse(norm_rec_file))
# normed_xml = transform(raw_xml)

tags = (
        "{https://www.fpds.gov/FPDS}award",
        "{https://www.fpds.gov/FPDS}IDV",
        "{https://www.fpds.gov/FPDS}OtherTransactionAward",
        "{https://www.fpds.gov/FPDS}OtherTransactionIDV",
)

for _, r in et.iterparse(xml_file, tag=tags):
    print(et.tostring(transform(r)))
    r.clear()
```

**Comment by spitfiredd** [comment]
I think you have given me the nudge need to convert this to python, but the docker script would also be nice.

**Comment by wamuir** [comment]
@spitfiredd I've added (see linked PR), want to give it a try before I merge?

**Comment by spitfiredd** [comment]
@wamuir - I can, but I can't get to it until Monday.

**Comment by wamuir** [comment]
Cool, no rush

**Comment by spitfiredd** [comment]
@wamuir - just getting back to this, do you know the docker command that will run this but the input/output are in a local drive?

**Comment by spitfiredd** [comment]
Just did:

`docker run -it -v ${PWD}:/tmp/data fpds-conversion-utility:1.0 fpds-conversion-utility/build/src/conversion-utility tmp/data/3355-NATIONALGALLERYOFART-AWARD.xml tmp/data/test.db`

It creates a test.db file on my computer.; and it looks like everything checks out.

Might want to consider making `fpds-conversion-utility/build/src/conversion-utility` the entrypoint?

I can send you a copy of the database if you like so you can compare?
